src/plugins/CubeSpawner.ts

```diff
@@ -45,14 +46,18 @@ export const CubeSpawnerModule = defineModule<GameState, CubeSpawnerModuleState>
     const module = getModule(ctx, CubeSpawnerModule);
 
     const image = createRemoteImage(ctx, { name: "Crate Image", uri: "/image/crate.gif" });
+    addResourceRef(ctx, image.resourceId);
```

Comment:
is there a reason we aren't calling `addResourceRef` in the resource creation functions themselves?

Reply:
We do when you attach it to a node. But if you then dispose that node and the reference count goes to 0 then the attached resources get cleaned up. This is acting as a second reference so that when the scene gets disposed, these resources stick around.

Reply:
ok sweet good to know ty!